### PR TITLE
TECH-114: Create feature implementation for Cucumber tests for database list and database schema read

### PR DESCRIPTION
**Pull request by mgrochalskisoldevelo**
Create feature implementation for Cucumber tests for database list and database schema read

## Description
For database list and database schema read features file created, implement the Cucumber features using javascript. 
Create all necessary code to implement the test that have been defined. Store the files in /tests/features/support.

TICKET https://govstack-global.atlassian.net/browse/TECH-114

[TECH-114]: https://govstack-global.atlassian.net/browse/TECH-114?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: test/gerkin/features/steps/db_datasets.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  'The user triggers an action with a valid payload to display information about Digital Registries databases', () => {
+    specDataRead
+      .get(`${baseUrl}`)
+      .withHeaders(`${header.key}`, `${header.value}`);
+  }
+);
+
+Then('The user received a list of Digital Registries databases', async () => {
+  await specDataRead.toss();
+  specDataRead.response().should.have.status(200);
```
Comment (KarolinaKopacz) [inline test/gerkin/features/steps/db_datasets.js:31]: In the titile is "The user received a list of..." - is it possible to check if user really receive those list? What is API response?

File: test/gerkin/features/steps/db_datasets.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  () => {
+    specDataRead
+      .get(`${baseUrl}`)
+      .withHeaders(`${header.key}`, '');
+  }
+);
+
+Then('The operation to receive information for Digital Registries databases results with wrong mediator error', async () => {
+  await specDataRead.toss();
+  specDataRead.response().should.have.status(400);
```
Comment (KarolinaKopacz) [inline test/gerkin/features/steps/db_datasets.js:46]: Is the response from API empty or there is some json or string ? If yes, check if those response is what you expect :)

File: test/gerkin/features/steps/db_datasets.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+When(
+  'The user triggers an action without a payload to display information about Digital Registries databases', () => {
+  specDataRead
+    .get(`${baseUrl}`);
+  }
+);
+  
+Then('The operation to receive information for Digital Registries databases results with no mediator error', async () => {
+  await specDataRead.toss();
+  specDataRead.response().should.have.status(400);
```
Comment (KarolinaKopacz) [inline test/gerkin/features/steps/db_datasets.js:60]: The same what above

File: test/gerkin/features/steps/db_datasets.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
+const pactum = require('pactum');
```
Comment (KarolinaKopacz) [inline test/gerkin/features/steps/db_datasets.js:1]: There is no `db_datasets.feature` file so please change the name of file to the `db_list.js`

File: test/gerkin/features/steps/db_read.js [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    specDataRead
+      .get(`${baseUrl}`)
+      .withHeaders(`${header.key}`, `${header.value}`)
+      .withPathParams('id', 353);
+  }
+);
+
+Then('The user received one database information with schema versions', async () => {
+  await specDataRead.toss();
+  specDataRead.response().should.have.status(200);
```
Comment (KarolinaKopacz) [inline test/gerkin/features/steps/db_read.js:32]: Is the response from API empty or there is some json or string ? If yes, check if those response is what you expect :)

File: test/gerkin/features/steps/db_read.js [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+  () => {
+    specDataRead
+      .get(`${baseUrl}`)
+      .withHeaders(`${header.key}`, `${header.value}`);
+  }
+);
+
+Then('The operation to receive any Digital Registries database information with schema versions results with no route param error', async () => {
+  await specDataRead.toss();
+  specDataRead.response().should.have.status(400);
```
Comment (KarolinaKopacz) [inline test/gerkin/features/steps/db_read.js:47]: Is the response from API empty or there is some json or string ? If yes, check if those response is what you expect :)

File: test/gerkin/features/steps/db_read.js [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+When(
+  'The user triggers an action with an invalid payload to display one database schema versions', () => {
+  specDataRead
+    .get(`${baseUrl}`);
+  }
+);
+  
+Then('The operation to receive any Digital Registries database information with schema versions results with wrong mediator error', async () => {
+  await specDataRead.toss();
+  specDataRead.response().should.have.status(400);
```
Comment (KarolinaKopacz) [inline test/gerkin/features/steps/db_read.js:61]: Is the response from API empty or there is some json or string ? If yes, check if those response is what you expect :)

File: test/gerkin/features/steps/db_read.js [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+When(
+    'The user triggers an action without a payload to display one database schema versions', () => {
+    specDataRead
+      .get(`${baseUrl}`);
+    }
+  );
+    
+  Then('The operation to receive any Digital Registries database information with schema versions results with no mediator error', async () => {
+    await specDataRead.toss();
+    specDataRead.response().should.have.status(400);
```
Comment (KarolinaKopacz) [inline test/gerkin/features/steps/db_read.js:74]: Is the response from API empty or there is some json or string ? If yes, check if those response is what you expect :)

File: test/gerkin/package.json [hunk 9]
```diff
@@ -5,7 +5,7 @@
     "test": "cucumber-js"
   },
   "devDependencies": {
-    "@cucumber/cucumber": "^8.9.1",
+    "@cucumber/cucumber": "^8.10.0",
```
Comment (KarolinaKopacz) [inline test/gerkin/package.json:8]: Is it necessary?

File: test/gerkin/features/steps/db_datasets.js [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+  'The user wants to display information about all Digital Registries databases', () => {
+  return 'The user wants to display information about all Digital Registries databases';
+  }
+);
+
+// Scenario: The user receives information about Digital Registries databases
+When(
+  'The user triggers an action with a valid payload to display information about Digital Registries databases', () => {
+    specDataRead
+      .get(`${baseUrl}`)
```
Comment (KarolinaKopacz) [inline test/gerkin/features/steps/db_datasets.js:24]:
```suggestion
      .get(baseUrl)
```

File: test/gerkin/features/steps/db_datasets.js [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+  await specDataRead.toss();
+  specDataRead.response().should.have.status(200);
+});
+
+// Scenario: The user is not able to receive information for Digital Registries databases because of empty value in the header
+When(
+  'The user triggers an action with an invalid payload to display information about Digital Registries databases',
+  () => {
+    specDataRead
+      .get(`${baseUrl}`)
```
Comment (KarolinaKopacz) [inline test/gerkin/features/steps/db_datasets.js:39]:
```suggestion
      .get(baseUrl)
```

File: test/gerkin/features/steps/db_datasets.js [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+  await specDataRead.toss();
+  specDataRead.response().should.have.status(400);
+});
+
+
+// Scenario: The user is not able to receive information for Digital Registries databases because of not including header
+When(
+  'The user triggers an action without a payload to display information about Digital Registries databases', () => {
+  specDataRead
+    .get(`${baseUrl}`);
```
Comment (KarolinaKopacz) [inline test/gerkin/features/steps/db_datasets.js:54]:
```suggestion
    .get(baseUrl);
```

File: test/gerkin/features/steps/db_read.js [hunk 13]
```diff
@@ -0,0 +1,79 @@
+const pactum = require('pactum');
+const { Given, When, Then, Before, After } = require('@cucumber/cucumber');
+const { header, localhost } = require('./helpers/helpers');
+
+let specDataRead;
+
+const baseUrl = `${localhost}database/{id}`;
+  
+Before(() => {
+  specDataRead = pactum.spec();
```
Comment (KarolinaKopacz) [inline test/gerkin/features/steps/db_read.js:10]:
```suggestion
  specDatabaseRead = pactum.spec();
```

File: test/gerkin/features/steps/db_read.js [hunk 14]
```diff
@@ -0,0 +1,79 @@
...
+  'The user wants to retrieve the Digital Registries database information with schema versions', () => {
+  return 'The user wants to retrieve the Digital Registries database information with schema versions';
+  }
+);
+
+// Scenario: The user successfully receives one Digital Registries database information with schema versions
+When(
+  'The user triggers an action with a valid payload to display one database schema versions', () => {
+    specDataRead
+      .get(`${baseUrl}`)
```
Comment (KarolinaKopacz) [inline test/gerkin/features/steps/db_read.js:24]:
```suggestion
      .get(baseUrl)
```

File: test/gerkin/features/steps/db_read.js [hunk 15]
```diff
@@ -0,0 +1,79 @@
...
+  await specDataRead.toss();
+  specDataRead.response().should.have.status(200);
+});
+
+// Scenario: The user is not able to receive any Digital Registries database information with schema versions bacause of not providing route param
+When(
+  'The user triggers an action without a route param to display one database schema versions',
+  () => {
+    specDataRead
+      .get(`${baseUrl}`)
```
Comment (KarolinaKopacz) [inline test/gerkin/features/steps/db_read.js:40]:
```suggestion
      .get(baseUrl)
```

File: test/gerkin/features/steps/db_read.js [hunk 16]
```diff
@@ -0,0 +1,79 @@
...
+  await specDataRead.toss();
+  specDataRead.response().should.have.status(400);
+});
+
+
+// Scenario: The user is not able to receive any Digital Registries database information with schema versions because of empty value in the header
+When(
+  'The user triggers an action with an invalid payload to display one database schema versions', () => {
+  specDataRead
+    .get(`${baseUrl}`);
```
Comment (KarolinaKopacz) [inline test/gerkin/features/steps/db_read.js:55]:
```suggestion
    .get(baseUrl);
```

File: test/gerkin/features/steps/db_read.js [hunk 17]
```diff
@@ -0,0 +1,79 @@
...
+Then('The operation to receive any Digital Registries database information with schema versions results with wrong mediator error', async () => {
+  await specDataRead.toss();
+  specDataRead.response().should.have.status(400);
+});
+
+// Scenario: The user is not able to receive any Digital Registries database information with schema versions because of not including header
+When(
+    'The user triggers an action without a payload to display one database schema versions', () => {
+    specDataRead
+      .get(`${baseUrl}`);
```
Comment (KarolinaKopacz) [inline test/gerkin/features/steps/db_read.js:68]:
```suggestion
      .get(baseUrl);
```

File: test/gerkin/features/steps/db_datasets.js [hunk 18]
```diff
@@ -0,0 +1,65 @@
+const pactum = require('pactum');
+const { Given, When, Then, Before, After } = require('@cucumber/cucumber');
+const { header, localhost } = require('./helpers/helpers');
+
+let specDataRead;
+
+const baseUrl = `${localhost}databases`;
+
+Before(() => {
+  specDataRead = pactum.spec();
```
Comment (KarolinaKopacz) [inline test/gerkin/features/steps/db_datasets.js:10]:
```suggestion
  specDatabaseList = pactum.spec();
```

**Review by KarolinaKopacz — CHANGES_REQUESTED** [review]

File: test/gerkin/features/db_list.feature [hunk 19]
```diff
@@ -10,8 +10,8 @@ Feature: API endpoint allowing users to get information about Digital Registries
 
     Scenario: The user is not able to receive information for Digital Registries databases because of empty value in the header
       When The user triggers an action with an invalid payload to display information about Digital Registries databases
-      Then The operation to receive information for Digital Registries databases results with wrong mediator error
+      Then The operation of receiving information about the databases of Digital Registers results with an error of the invalid header
```
Comment (KarolinaKopacz) [inline test/gerkin/features/db_list.feature:13]:
```suggestion
      The result of an operation to receive information about the databases of Digital Registers returns an error
```

File: test/gerkin/features/steps/db_list.js [hunk 20]
```diff
@@ -0,0 +1,167 @@
+const pactum = require('pactum');
+const { Given, When, Then, Before, After } = require('@cucumber/cucumber');
+const { header, localhost } = require('./helpers/helpers');
+
+let specDatabaseRead;
```
Comment (KarolinaKopacz) [inline test/gerkin/features/steps/db_list.js:5]:
```suggestion
let specDatabaseList;
```

File: test/gerkin/features/steps/db_list.js [hunk 21]
```diff
@@ -0,0 +1,167 @@
...
+      "has_logo": false
+    }
+  ],
+  "group_id": 10,
+  "order": 76,
+  "data_index_increment": 3
+}
+
+Before(() => {
+  specDatabaseRead = pactum.spec();
```
Comment (KarolinaKopacz) [inline test/gerkin/features/steps/db_list.js:110]:
```suggestion
  specDatabaseList = pactum.spec();
```

**Comment by KarolinaKopacz** [comment]
Great! You added a lot of changes.

**Review by conradsp — APPROVED** [review]

**Review by KarolinaKopacz — APPROVED** [review]

**Review by dborowiecki — APPROVED** [review]

**Pull request merged**